docs: document library inclusion criteria (#304)
canonical/charmlibs: .docs/how-to/python-package.md

Instead, consider including such libraries in a shared team package for common code, distributing them as a `git` dependency, or including them as a local dependency (in a charm monorepo).
If none of those options work for you, then you might consider publishing them as generic packages on PyPI.

Read on to learn more about your options for distributing Python packages.

(charmlibs-inclusion)=

> ## What packages should be published under the charmlibs namespace?

Reviewer: How about shortening the title slightly?

```suggestion
## What can I publish under the charmlibs namespace?
```

The `charmlibs` namespace should only be used by packages published from the `charmlibs` monorepo.
But what packages should be included in this monorepo?

`charmlibs` is for libraries that are likely to be useful to a wide range of charms.
- For {ref}`general libraries <charm-libs-general>`, this means well-scoped libraries providing functionality that would be useful to multiple different charms (or libraries).
Shared code for specific teams' patterns (or specific workloads) should instead be distributed by one of the other methods outlined in this document.
- For {ref}`interface libraries <charm-libs-interface>`, this means libraries for public interfaces intended to be used by many different charms. This doesn't include interfaces that are intended for communication between (for example) a pair of tightly coupled charms.

New libraries should have a specification with some cross-team buy-in before being added to `charmlibs`.
Ideally, they would implement patterns that have already been tested in production.

> One path to publication that achieves this would be for charmers to develop new libraries as modules of their team-specific packages, use them in their charms, and then migrate them to `charmlibs` when they prove useful.

Reviewer: What do you think about turning this into a recommendation? Something like:

```suggestion
For new libraries, we recommend that you start by developing a module inside a team-specific package, then migrate the module to `charmlibs` after using it in your charms.
```

Author: I'm not opposed to starting with a spec and going straight to `charmlibs` in principle, so how about this:
```suggestion
For example, charmers could develop a new library as a sub-package of their team-specific packages, use it in their charms, and then migrate it to `charmlibs` when it proves useful in production.
```
